java-components/build-recipes-database/src/main/java/com/redhat/hacbs/recipies/build/CustomDeserialiser.java

```diff
@@ -0,0 +1,45 @@
+package com.redhat.hacbs.recipies.build;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+public class CustomDeserialiser
+        extends StdDeserializer<Map<String, BuildRecipeInfo>> {
+
+    public CustomDeserialiser() {
+        this(null);
+    }
+
+    public CustomDeserialiser(Class<BuildRecipeInfo> vc) {
+        super(vc);
+    }
+
+    @Override
+    public Map<String, BuildRecipeInfo> deserialize(JsonParser p, DeserializationContext ctxt)
+            throws IOException {
+        BuildRecipeInfo parentBuild = (BuildRecipeInfo) p.getParsingContext().getParent().getCurrentValue();
```

Comment:
I don't really understand this merge logic? 

I am guessing it is trying to use the parent values and merge them in, but AFAIK there is no guarantee the parent is fully constructed at this point? (i.e. do you get different outcomes based on where the additionalBuilds field appears in the YAML)?

Personally I don't think we need this, if some info needs to be copied it is not a big deal, but it has the advantage of making the build recipe be very explicit.

Reply:
Ok, hadn't seen that about parent construction which means it was order dependent - I'll remove that and update the doc in the upcoming pr for the jvm-build-data.